web_poet/functional.py

```diff
@@ -0,0 +1,52 @@
+from typing import Callable, Type
+
+from web_poet.pages import ItemPage
+
+
+def page_object(func: Callable) -> Type[ItemPage]:
+    """
+    Decorator to create a Page Object from a function.
+    Note that the original function is destroyed, and
+    a class is created instead.
+
+    .. code-block:: python
+
+        @web_poet.page_object
+        def MyPage(resp: HttpResponse):
+            return {"title": resp.css("title::text").get()}
+
+    is a shortcut for
+
+    .. code-block:: python
+
+        class MyPage(web_poet.ItemPage):
+            def __init__(self, resp: HttpResponse):
+                self.resp = resp
+
+            def to_item(self):
+                return {"title": self.resp.css("title::text").get()}
+
+    """
+    class PageObject(ItemPage):
+        def __init__(self, *args, **kwargs):
+            # FIXME: save arguments as properly named attributes
```

Comment:
Could you elaborate?

Reply:
This is optional, to keep the generated class as close as possible to the hand-written class. See example above - MyPage has self.resp attribute.